### ISSUE TITLE
amazon-image.nix does not take in account for NVMe-only EC2

### ISSUE THREAD

**Issue by RaitoBezarius** [NixOS/nixpkgs#72274]
**Describe the bug**
If you have NVMe-only EC2, the GRUB installation will fail.

**To Reproduce**
1. Take an NVMe-only EC2 instance, a t3a.large for example.
2. Try to `nixops deploy` something to it.
3. GRUB install will fail.

**Expected behavior**
GRUB installation should succeed.

**Additional context**
`amazon-image.nix` hardcodes `/dev/xvda` as device, when it's possible it is only `/dev/nvme0n1`.

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.7-arch1-1-ARCH, Arch Linux, noversion`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(root): `"nixpkgs-20.03pre197736.91d5b3f07d2"`
 - channels(raito): `"nixpkgs-20.03pre197736.91d5b3f07d2"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```
Maintainer information:
```yaml
module: amazon-image
```

**Comment by thefloweringash** [comment]
Related #67349

**Comment by nh2** [comment]
This seems be fixed for me with PR #86060 when I set `nixosAmiVersion = "20.03";` (n option I added in https://github.com/NixOS/nixops/pull/972/commits/eacd6000b2c42b9b52e49f972be8f73e74004c59 but it hasn't made it into `nixops` yet).

**Comment by nh2** [comment]
> This seems be fixed for me with PR #86060 when I set `nixosAmiVersion = "20.03";`

Closing given that I think it's fixed in 20.03; reopen if that's false.
